puma/metrics.py

```diff
@@ -5,11 +5,45 @@
 from puma.utils.histogram import hist_w_unc, save_divide
 
 
+def weighted_percentile(
+    data: np.ndarray,
+    perc: np.ndarray,
+    weights: np.ndarray = None,
+):
+    """Calculate weighted percentile.
+
+    Parameters
+    ----------
+    data : np.ndarray
+        Data array
+    perc : np.ndarray
+        Percentile array
+    weights : np.ndarray
+        Weights array, by default None
+
+    Returns
+    -------
+    np.ndarray
+        Weighted percentile array
+    """
+    if weights is None:
+        weights = np.ones_like(data)
+    ix = np.argsort(data)
+    data = data[ix]  # sort data
+    weights = weights[ix]  # sort weights
+    cdf = np.cumsum(weights) - 0.5 * weights
+    cdf -= cdf[0]
```

Comment:
I'm also unclear on why you apply this, might be helpful to add some comments to these lines